analytics_data_api/v0/models.py

```diff
@@ -445,7 +445,9 @@ class ModuleEngagementMetricRanges(models.Model):
     low_value and high_value.
     """
 
-    course_id = models.CharField(db_index=True, max_length=255)
+    # primary key is actually course_id, metric, and range_type, but only one
+    # primary is allowed in django
+    course_id = models.CharField(primary_key=True, db_index=True, max_length=255)
```

Comment:
It is actually setup this way precisely because of the special properties of this type of system.

We are trading flexibility (which we will likely never need) for query performance (which we know we want).

The pipeline currently provides the following assurances:

* Any table can be regenerated from scratch at any point in time.
* The API is a read-only consumer of the database.

Given those invariants, we can make performance optimizations like this one. Frankly, the fact that we didn't do this with all of our other tables is somewhat of an oversight. Using a meaningful primary key like this dramatically increases query performance at the cost of inflexibility and write performance. Given that we write infrequently and read often I think this is the correct trade-off.

Django assumes your database is a traditional OLTP system. This database is really just a cache, so we can (and should) take advantage of those differences in usage patterns.